### PR TITLE
Parameterize image fixture for testing single-channel 2D, 3D and multichannel 2D

### PR DESCRIPTION
**Pull request by mcquin**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -3,15 +3,25 @@
...
 import skimage.data
 import pytest
 
 
-@pytest.fixture(scope="module")
-def image():
-    example = skimage.data.camera()
-
-    return cellprofiler.image.Image(example)
+@pytest.fixture(scope="module",
```
Comment (0x00b1) [inline tests/conftest.py]: Add a newline

File: tests/conftest.py [hunk 2]
```diff
@@ -3,15 +3,25 @@
...
+                params=[
+                    (skimage.data.camera()[0:128, 0:128], 2),
+                    (skimage.data.astronaut()[0:128, 0:128, :], 2),
+                    (numpy.tile(skimage.data.camera()[0:32, 0:32], (2, 1)).reshape(2, 32, 32), 3)
+                ],
+                ids=[
+                    "grayscale_2d",
+                    "multichannel_2d",
+                    "grayscale_3d"
+                ])
```
Comment (0x00b1) [inline tests/conftest.py]: Add a newline

File: tests/conftest.py [hunk 3]
```diff
@@ -3,15 +3,25 @@
...
+def image(request):
+    data, dimensions = request.param
+    return cellprofiler.image.Image(image=data, dimensions=dimensions)
```
Comment (0x00b1) [inline tests/conftest.py]: Add a newline

File: tests/conftest.py [hunk 4]
```diff
@@ -3,15 +3,25 @@
...
-
-    return cellprofiler.image.Image(example)
+@pytest.fixture(scope="module",
+                params=[
+                    (skimage.data.camera()[0:128, 0:128], 2),
+                    (skimage.data.astronaut()[0:128, 0:128, :], 2),
+                    (numpy.tile(skimage.data.camera()[0:32, 0:32], (2, 1)).reshape(2, 32, 32), 3)
+                ],
+                ids=[
+                    "grayscale_2d",
```
Comment (0x00b1) [inline tests/conftest.py]: `grayscale_image`, `multichannel_image`, and `grayscale_volume` are more descriptive.

File: tests/conftest.py [hunk 5]
```diff
@@ -3,15 +3,25 @@
...
 
-@pytest.fixture(scope="module")
-def image():
-    example = skimage.data.camera()
-
-    return cellprofiler.image.Image(example)
+@pytest.fixture(scope="module",
+                params=[
+                    (skimage.data.camera()[0:128, 0:128], 2),
+                    (skimage.data.astronaut()[0:128, 0:128, :], 2),
```
Comment (0x00b1) [inline tests/conftest.py]: Nice choice. Typically, I use [mandrill](http://graphics.cs.williams.edu/data/images/mandrill.png). But I like this better.

File: tests/conftest.py [hunk 6]
```diff
@@ -3,15 +3,25 @@
...
+                    (numpy.tile(skimage.data.camera()[0:32, 0:32], (2, 1)).reshape(2, 32, 32), 3)
```
Comment (0x00b1) [inline tests/conftest.py]: I think this is fine for the moment, but we should really get a volume upstream to skimage.

File: tests/test_image.py [hunk 7]
```diff
@@ -79,6 +79,27 @@ def test_init_volume(self):
 
         self.assertEqual(x.dimensions, 3)
 
+    def test_multichannel_grayscale_2d(self):
```
Comment (0x00b1) [inline tests/test_image.py]: Multi-channel grayscale?

File: tests/test_image.py [hunk 8]
```diff
@@ -79,6 +79,27 @@ def test_init_volume(self):
...
+        self.assertFalse(x.multichannel)
+
+    def test_multichannel_rgb_2d(self):
+        data = np.ones((10, 10, 3))
+
+        x = cpi.Image(image=data)
+
+        self.assertTrue(x.multichannel)
+
+    def test_multichannel_grayscale_3d(self):
```
Comment (0x00b1) [inline tests/test_image.py]: Multi-channel grayscale?

File: tests/test_image.py (hunk 7 above)
Comment (mcquin) [inline tests/test_image.py]: the name of the method under test is `multichannel`

Comment (mcquin) [inline tests/test_image.py]: the name of the method under test is `multichannel`

File: tests/test_image.py (hunk 8 above)
Comment (mcquin) [inline tests/test_image.py]: the name of the method under test is `multichannel`

**Comment by mcquin** [comment]
@0x00b1 ready for review

**Pull request merged**